include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -390,6 +468,315 @@ namespace alpaka
+                            + std::to_string(
+                                trait::GetAccDevProps<TAcc>::getAccDevProps(getDev(queue)).m_multiProcessorCount)
+                            + ".");
+                    }
+                }
+
+#        endif
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                {
+                    // Log maximum number of blocks that can be requested by this kernel.
+                    int numBlocksPerSm = 0;
+                    cudaOccupancyMaxActiveBlocksPerMultiprocessor(
+                        &numBlocksPerSm,
+                        kernelName,
+                        blockThreadExtent.prod(),
+                        static_cast<std::size_t>(blockSharedMemDynSizeBytes));
+                    auto multiProcessorCount
+                        = trait::GetAccDevProps<TAcc>::getAccDevProps(getDev(queue)).m_multiProcessorCount;
+                    std::cout << "maxBlocksPerGrid for the " << core::demangled<TKernelFnObj> << ": "
+                              << numBlocksPerSm * multiProcessorCount << std::endl;
+                }
+#        endif
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                // Log the function attributes.
+                typename TApi::FuncAttributes_t funcAttrs;
+                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(TApi::funcGetAttributes(&funcAttrs, kernelName));
+                std::cout << __func__ << " binaryVersion: " << funcAttrs.binaryVersion
+                          << " constSizeBytes: " << funcAttrs.constSizeBytes << " B"
+                          << " localSizeBytes: " << funcAttrs.localSizeBytes << " B"
+                          << " maxThreadsPerBlock: " << funcAttrs.maxThreadsPerBlock
+                          << " numRegs: " << funcAttrs.numRegs << " ptxVersion: " << funcAttrs.ptxVersion
+                          << " sharedSizeBytes: " << funcAttrs.sharedSizeBytes << " B" << std::endl;
+#        endif
+
+                // Set the current device.
+                ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(TApi::setDevice(queue.m_spQueueImpl->m_dev.getNativeHandle()));
+                // Enqueue the kernel execution.
+                // \NOTE: No const reference (const &) is allowed as the parameter type because the kernel launch
+                // language extension expects the arguments by value. This forces the type of a float argument given
+                // with std::forward to this function to be of type float instead of e.g. "float const & __ptr64"
+                // (MSVC). If not given by value, the kernel launch code does not copy the value but the pointer to the
+                // value location.
+                std::apply(
+                    [&](remove_restrict_t<std::decay_t<TArgs>> const&... args)
+                    {
+                        void const* kernelArgs[] = {&threadElemExtent, &task.m_kernelFnObj, &args...};
+                        cudaLaunchCooperativeKernel(
```

Comment:
eventually, can you add an entry for this function in `include/alpaka/core/ApiCudaRt.hpp` and `include/alpaka/core/ApiHipRt.hpp` ?